Implement image URL builder

Adoby7 opened the pull request:
### Motivation

Fix #67 


### Checklist

- [x] Code follows coding conventions held in this repo
- [x] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

The validation of parameters is delegated to Kentico cloud API.

The documentation generated by `jazzy` seems to be messy, but all of them are in `docs/` folder

makma commented:
Thank you @Adoby7 for your great effort! I'll review code later this week, probably on Friday.

Adoby7 commented on Example/Tests/DeliveryService/ImageUrlBuilder/ImageUrlBuilder.swift:
```diff
@@ -0,0 +1,299 @@
+                    let actualUrl = ImageUrlBuilder(baseUrl: baseUrl)
+                        .withRectangleCrop(x: 100, y: 50, width: 700, height: 400)
+                        .url
+                    let expectedUrl = baseUrl + "?rect=100.0,50.0,700,400"
```
Oh it seems that I misunderstood the specification in the past. Since each of 4 parameters can be either `float` or `int`, it has 16 combinations in total. I just implement two of them, which are all `int` case and all `float` case. I have updated the testing accordingly as well.

makma replied:
Honestly - I'm not sure as well about first two params and their float (percentage) values. I'll ask your technical writers and let you know.

Adoby7 commented:
Hi @makma, sorry for the typo. I have fixed them. By the way, I have updated the comments and documentation about the restricted range of each parameter, if applicable.

makma commented:
Thank you for implementing this feature and your pull request! Your work is appreciated. Just one note for the future - do not use force push on branches you share with somebody else. This also applies to [pull requests](https://help.github.com/en/articles/about-pull-requests). 

> When pushing commits to a pull request, don't force push. Force pushing can corrupt your pull request.

Thank you.
